frontend_tests/node_tests/message_events.js

```diff
@@ -57,31 +82,103 @@ function test_helper(side_effects) {
     return self;
 }
 
-run_test('update_messages', () => {
-    const original_message = {
-        id: 111,
-        display_recipient: denmark.name,
-        flags: ['mentioned'],
-        sender_id: alice.user_id,
-        stream_id: denmark.stream_id,
-        topic: 'lunch',
-        type: 'stream',
-    };
+run_test('update_topic', () => {
 
     message_store.add_message_metadata(original_message);
     message_store.set_message_booleans(original_message);
 
     assert.equal(original_message.mentioned, true);
     assert.equal(original_message.unread, true);
-
     assert.deepEqual(
         stream_topic_history.get_recent_topic_names(denmark.stream_id),
-        ['lunch']
+        ['lunch-old']
     );
 
     unread.update_message_for_mention(original_message);
     assert(unread.unread_mentions_counter.has(original_message.id));
 
+    const events = [
+        {
+            flags: [],
+            message_id: 111,
+            message_ids: [111],
+            orig_subject: 'lunch-old',
+            stream_id: denmark.stream_id,
+            subject: 'lunch',
+        },
+    ];
+
+    stream_data.get_sub_by_id = function (stream_id) {
+        assert(stream_id, 101);
+        return {name: denmark.name};
+    };
+
+    unread.update_unread_topics = function (msg, event) {
+        assert(msg.id, 111);
+        assert(event.message_id, 111);
+    };
+
+    current_msg_list.get_row = (message_id) => {
+        assert.equal(message_id, 111);
+        return ['row-stub'];
+    };
+
+    let rendered_msg;
+
+    alert_words.process_message = (msg) => {
+        rendered_msg = msg;
+    };
+
+    const helper = test_helper(side_effects);
+
+    page_params.realm_allow_edit_history = false;
+    message_list.narrowed = 'stub-to-ignore';
+
+    // TEST THIS:
+    message_events.update_messages(events);
+
+    assert(!unread.unread_mentions_counter.has(original_message.id));
+
+    helper.verify();
+
+    assert.deepEqual(rendered_msg,  {
+        alerted: false,
+        collapsed: false,
+        display_recipient: 'Denmark',
+        historical: false,
+        id: 111,
+        is_stream: true,
+        last_edit_timestamp: undefined,
+        mentioned: false,
+        mentioned_me_directly: false,
+        reactions: [],
+        reply_to: 'alice@example.com',
+        sender_email: 'alice@example.com',
+        sender_full_name: 'Alice Patel',
+        sender_id: 32,
+        sent_by_me: false,
+        starred: false,
+        stream: 'Denmark',
+        stream_id: 101,
+        topic: 'lunch',
+        topic_links: undefined,
+        type: 'stream',
+        unread: true,
+    });
+
+});
+
+run_test('update_messages', () => {
+
```

Comment:
Another whitespace anomaly -- our style is not to have blank lines at the start of a test.